File: livraria-backend/src/main/java/com/tqi/evolutionfullstack/livrariabackend/service/CustomerService.java
package com.tqi.evolutionfullstack.livrariabackend.service;

import com.tqi.evolutionfullstack.livrariabackend.model.Customer;
import java.util.List;
import java.util.Optional;

public interface CustomerService {

    Customer SaveCustomer(Customer customer);

    void deleteCustomer(Customer customer);

    List<Customer> listCustomer(Customer customerId);

    List<Customer> findAll();

    boolean EvaluateFields(Customer customer);

    Optional<Customer> getCustomerById(Long id);
}

File: livraria-frontend/src/services/customerService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/customer";

class CustomerService{

    
    addCustomer(customer){
        return axios.post(BASE_URL, customer);
    }

    getCustomer(){
        return axios.get(BASE_URL);
    }

    getCustomerById(customerId){
        return axios.get(BASE_URL + '/' + customerId);
    }

    deleteCustomer(customerId){
        return axios.delete(BASE_URL + '/' + customerId);
    }
    
}

export default new CustomerService()